gulp/bower.js

```diff
@@ -0,0 +1,84 @@
+var gulp = require('gulp');
+var plugins = require('gulp-load-plugins')();
+var sequence = plugins.sequence.use(gulp);
+
+var fs = require('fs');
+var del = require('del');
+var path = require('path');
+var deploy = require('gh-pages');
+
+var env = process.env;
+var pkg = require('../package.json');
+
+var out = {
+    bower: './out/bower',
+    repo: './out/bower-repo'
+};
+
+gulp.task('bower', sequence(
+    'clean-bower-folder',
+    ['copy-license', 'copy-bower-json', 'copy-bower-artifacts', 'copy-readme'],
+    'publish-bower-repo'
+));
+
+gulp.task('clean-bower-folder', [], function () {
+    return del([out.bower, out.repo]);
+});
+
+gulp.task('copy-license', [], function () {
+    return gulp.src('./license.md').pipe(gulp.dest(out.bower));
+});
+
+gulp.task('copy-bower-artifacts', [], function () {
+    var distFiles = './out/dist/**/*';
+    return gulp.src(distFiles).pipe(gulp.dest(out.bower));
+});
+
+gulp.task('copy-bower-json', [], function () {
+    var bowerJsonPath = './publish/bower.json';
+    return gulp.src(bowerJsonPath)
+        .pipe(plugins.template(pkg))
+        .pipe(gulp.dest(out.bower));
+});
+
+gulp.task('copy-readme', [], function () {
+    return gulp.src('./README.md')
+        .pipe(_addHeader('./publish/readme.header.md'))
+        .pipe(gulp.dest(out.bower));
+});
+
+gulp.task('publish-bower-repo', [], function (cb) {
+
+    // todo: maybe should be moved to appveyor.yml
+    var bowerRepoUrl = _format('https://{identity}:{authToken}@{repo}', {
+        identity: env.BOWER_REPO_PUBLISH_IDENTITY,
+        authToken: env.BOWER_REPO_PUBLISH_TOKEN,
+        repo: env.BOWER_REPO
+    });
+
+    var commitMessage = _format('{version} version was published', {version: env.APPVEYOR_REPO_TAG_NAME});
+
+    deploy.publish(path.join(__dirname, '../out/bower'), {
+        repo: bowerRepoUrl,
+        branch: env.BOWER_BRANCH,
+        remote: 'origin',
+        tag: env.APPVEYOR_REPO_TAG_NAME,
+        clone: path.join(__dirname, '../out/bower-repo'),
+        message: commitMessage,
+        silent: true
+    }, cb);
+});
+
+function _addHeader(fileName) {
```

Comment:
Good habit, but lets rename them to use the same code style (not formalized yet for js files - it needs to be done at some moment - jslint?)